src/main/java/io/jenkins/plugins/oidc_provider/config/ClaimTemplate.java

```diff
@@ -37,20 +37,36 @@
 import org.kohsuke.accmod.Restricted;
 import org.kohsuke.accmod.restrictions.NoExternalUse;
 import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 import org.kohsuke.stapler.QueryParameter;
 
 public final class ClaimTemplate extends AbstractDescribableImpl<ClaimTemplate> {
 
     public final @NonNull String name;
     public final @NonNull String format;
     public final @NonNull ClaimType type;
-    public final String requiredEnvVars;
+    private String requiredEnvVars;
 
-    @DataBoundConstructor public ClaimTemplate(String name, String format, ClaimType type, String requiredEnvVars) {
+    public ClaimTemplate(String name, String format, ClaimType type) {
         this.name = name;
         this.format = format;
         this.type = type;
-        this.requiredEnvVars = requiredEnvVars;
+    }
+
+    @DataBoundConstructor public ClaimTemplate(String name, String format, ClaimType type, String vars) {
```

Comment:
A field configured via `@DataBoundSetter` should _not_ also be configured by `@DataBoundConstructor`, so you can revert constructor changes.

Reply:
is it ok to keep the new constructor without the `@DataBoundConstructor` for the unit tests?

Reply:
If you like. I usually do not since it seems more confusing.